src/jit/gentree.cpp

```diff
@@ -1740,6 +1740,35 @@ bool GenTreeCall::IsHelperCall(Compiler* compiler, unsigned helper) const
     return IsHelperCall(compiler->eeFindHelper(helper));
 }
 
+void GenTreeCall::ReplaceCallOperand(GenTree** operandUseEdge, GenTree* replacement)
+{
+    assert(operandUseEdge != nullptr);
+    assert(replacement != nullptr);
+    assert(TryGetUse(*operandUseEdge, &operandUseEdge));
+
+    GenTree* originalOperand = *operandUseEdge;
+    *operandUseEdge = replacement;
+
+    const bool isArgument = (replacement != gtControlExpr) && ((gtCallType != CT_INDIRECT) ||
+        ((replacement != gtCallCookie) && (replacement != gtCallAddr)));
+
+    if (isArgument)
+    {
+        if ((originalOperand->gtFlags & GTF_LATE_ARG) != 0)
+        {
+            replacement->gtFlags |= GTF_LATE_ARG;
+        }
+        else
+        {
+            assert((replacement->gtFlags & GTF_LATE_ARG) == 0);
+
+            fgArgTabEntryPtr fp = Compiler::gtArgEntryByNode(this, originalOperand);
+            assert(fp->node == originalOperand);
+            fp->node = replacement;
```

Comment:
You should consider preserving the Jit dump information that the former implementation had when a mutation event occurs.

JITDUMP("parent call was :\n");